### PR TITLE
[GLUTEN-5448][VL] Fix the issue of cleaning when left with previous build in velox-backends part

### PR DESCRIPTION
**Pull request by Donvi**
## What changes were proposed in this pull request?

Just add sudo command to the clean step to make sure all of those files removed

## How was this patch tested?

mannual test in local build and find it can work well

### CODE REVIEW

File: dev/builddeps-veloxbe.sh [hunk 1]
```diff
@@ -189,7 +189,7 @@ function build_velox {
 function build_gluten_cpp {
   echo "Start to Gluten CPP"
   cd $GLUTEN_DIR/cpp
-  rm -rf build
+  sudo rm -rf build
   mkdir build
   cd build
```
Comment (Yohahaha) [inline dev/builddeps-veloxbe.sh:194]: do we really need this change? if you need sudo rm, you will need sudo mkdir too, could you double check it?

File: ep/build-velox/src/build_velox.sh [hunk 2]
```diff
@@ -209,7 +209,7 @@ function check_commit {
     fi
   else
     # Branch-new build requires all untracked files to be deleted. We only need the source code.
-    git clean -dffx :/
+    sudo git clean -dffx :/
```
Comment (Yohahaha) [inline ep/build-velox/src/build_velox.sh:212]: great catch!

File: dev/builddeps-veloxbe.sh [hunk 3]
```diff
@@ -213,4 +213,4 @@ commands_to_run=${OTHER_ARGUMENTS:-}
        "${cmd}"
     done
   fi
-)
\ No newline at end of file
+)
```
Comment (Yohahaha) [inline dev/builddeps-veloxbe.sh:216]: just avoid introducing unnecessary changes.

**Review by Yohahaha — COMMENTED** [review]
thank you! some minor comments

File: dev/builddeps-veloxbe.sh (hunk 1 above)
Comment (Donvi) [reply dev/builddeps-veloxbe.sh]: Thanks for reminding, fixed.  But this is only for removing, so this should not influence the build result.

**Review by Yohahaha — APPROVED** [review]
LGTM!
